include/layers/SoftmaxLayer.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <algorithm>
+#include <cmath>
+#include <stdexcept>
+#include <vector>
+
+#include "layers/Layer.hpp"
+#include "layers/Tensor.hpp"
+
+namespace it_lab_ai {
+
+class SoftmaxLayer : public Layer {
+ public:
+  explicit SoftmaxLayer(int axis = -1) : axis_(axis) {}
```

Comment:
Add tests for negative/large `axis`, axis normalization, and numerical stability (subtract max to prevent overflow).